### PR TITLE
do not close channels in function conatinerLog

### PR DESCRIPTION
**Pull request by ydayagi**
it is better to avoid closing channels

https://issues.redhat.com/browse/ECOPROJECT-837
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

File: internal/workload/podman/podman.go [hunk 1]
```diff
@@ -515,8 +515,6 @@ func (p *podman) containerLog(ctx context.Context, containerID string, res io.Wr
 					log.Errorf("cannot write log line to io.Writer for container '%v': %v", containerID, err)
 				}
 			case <-ctx.Done():
-				close(stderrCh)
```
Comment (tupyy) [inline internal/workload/podman/podman.go:518]: I think we should close the producer goroutine also. If you close only this one, data keeps accumulating into the channel for nothing.

Comment (ydayagi) [reply internal/workload/podman/podman.go:518]: data into the channel is not a problem. but if no one closes the routine then we may have a leak. I will check it.

Comment (ydayagi) [reply internal/workload/podman/podman.go:518]: @tupyy I opened a separate bug for this issue: https://issues.redhat.com/browse/ECOPROJECT-842

**Review by masayag — APPROVED** [review]

**Review by jakub-dzon — APPROVED** [review]

File: internal/workload/podman/podman.go (hunk 1 above)
Comment (tupyy) [reply internal/workload/podman/podman.go:518]: @jakub-dzon yes

**Pull request merged**